ion-schema/src/isl/isl_constraint.rs

```diff
@@ -730,6 +741,29 @@ impl IslConstraintImpl {
                     }
                 }
             }
+            "ieee754_float" => {
+                if value.is_null() {
+                    return invalid_schema_error(
+                        "expected a symbol for an `ieee754_float` constraint, found null",
+                    );
+                }
+                if !value.annotations().is_empty() {
+                    return invalid_schema_error(
+                        "`ieee_754_float` constraint must be one of binary16, binary32, binary64",
```

Comment:
The message is a little vague for the condition that's being checked; maybe something like this?
```suggestion
                        "`ieee_754_float` argument must not have annotations ",
```